packages/amplication-git-utils/src/git/github.service.ts

```diff
@@ -3,31 +3,38 @@ import { ConfigService } from "@nestjs/config";
 import { createAppAuth } from "@octokit/auth-app";
 import { components } from "@octokit/openapi-types";
 import { App, Octokit } from "octokit";
-import { createPullRequest } from "octokit-plugin-create-pull-request";
-import { Changes } from "octokit-plugin-create-pull-request/dist-types/types";
 import {
   EnumGitOrganizationType,
   EnumPullRequestMode,
+  GitClient,
   GithubFile,
+  Options,
+  PullRequestModule,
   RemoteGitOrganization,
   RemoteGitRepos,
   RemoteGitRepository,
 } from "../types";
 import { ConverterUtil } from "../utils/convert-to-number";
-import { UNSUPPORTED_GIT_ORGANIZATION_TYPE } from "./git.constants";
+import {
+  AMPLICATION_IGNORED_FOLDER,
+  UNSUPPORTED_GIT_ORGANIZATION_TYPE,
+} from "./git.constants";
 import { AccumulativePullRequest } from "./github/AccumulativePullRequest";
 import { BasicPullRequest } from "./github/BasicPullRequest";
+import { createPullRequest } from "octokit-plugin-create-pull-request";
+import { AmplicationIgnoreManger } from "../utils/amplication-ignore-manger";
+import { Changes } from "octokit-plugin-create-pull-request/dist-types/types";
+import { join } from "lodash";
 
 const GITHUB_FILE_TYPE = "file";
 export const GITHUB_CLIENT_SECRET_VAR = "GITHUB_CLIENT_SECRET";
 export const GITHUB_APP_APP_ID_VAR = "GITHUB_APP_APP_ID";
 export const GITHUB_APP_PRIVATE_KEY_VAR = "GITHUB_APP_PRIVATE_KEY";
 export const GITHUB_APP_INSTALLATION_URL_VAR = "GITHUB_APP_INSTALLATION_URL";
 export const UNEXPECTED_FILE_TYPE_OR_ENCODING = `Unexpected file type or encoding received`;
-
 type DirectoryItem = components["schemas"]["content-directory"][number];
-@Injectable()
-export class GithubService {
+
+export class GithubService implements GitClient {
```

Comment:
this service should only be dependent on the Git provider API and should not be dependent on nest.
All the configuration should be provided as parameters to the functions instead of using the config service